dash/dash-renderer/src/actions/callbacks.ts

```diff
@@ -324,19 +324,37 @@ async function handleClientside(
     return result;
 }
 
-function sideUpdate(outputs: any, dispatch: any, paths: any) {
-    toPairs(outputs).forEach(([id, value]) => {
-        const [componentId, propName] = id.split('.');
-        const componentPath = paths.strs[componentId];
+function updateComponent(component_id: any, props: any) {
+    return function (dispatch: any, getState: any) {
+        const paths = getState().paths;
+        const componentPath = getPath(paths, component_id);
         dispatch(
             updateProps({
-                props: {[propName]: value},
+                props,
                 itempath: componentPath
             })
         );
-        dispatch(
-            notifyObservers({id: componentId, props: {[propName]: value}})
-        );
+        dispatch(notifyObservers({id: component_id, props}));
+    };
+}
+
+function sideUpdate(outputs: any, dispatch: any) {
+    toPairs(outputs).forEach(([id, value]) => {
+        let componentId, propName;
+        if (id.includes('.')) {
+            [componentId, propName] = id.split('.');
+            if (componentId.startsWith('{')) {
+                componentId = JSON.parse(componentId);
+            }
+            dispatch(updateComponent(componentId, {[propName]: value}));
+        } else {
+            if (id.startsWith('{')) {
+                componentId = JSON.parse(id);
+            } else {
+                componentId = id;
+            }
+            dispatch(updateComponent(componentId, value));
+        }
```

Comment:
I think we could remove some of the checks and method calls that happens twice if we do something like:

```js
  toPairs(outputs).forEach(([id, value]) => {
    let componentId, propName = value;
    if (id.includes('.')) {
      [componentId, propName] = id.split('.');
    } else {
      componentId = id;
    }

    if (componentId.startsWith('{')) {
      componentId = JSON.parse(componentId);
    }

    const props = propName === value ? value : {[propName]: value};
    dispatch(updateComponent(componentId, props));
  });
```

We always parse if it starts by `{` and if we split then we know the propName is not equal to `value` from the `let`.